src/main/java/frc/robot/subsystems/internals/InternalSubsystem.java

```diff
@@ -101,11 +101,17 @@ public InternalSubsystem(TurretSubsystem turretSubsystem) {
     public void periodic() {
         // Check sensors for incoming and exiting balls and update ball counts accordingly
         boolean entranceDetected = entrance.get() >= 0.4;
-        if (!prevEntranceDetected && entranceDetected) entranceStorageBallCount++;
+        if (!prevEntranceDetected && entranceDetected) {
+            entranceStorageBallCount++;
+            System.out.println("entrance ball detected");
+            //TODO shouldn't this set rejectingChecked to false?
```

Comment:
It should not, because we only want to tell the turret to reject based on the color of the _first_ ball. That's why `rejectingChecked` exists in the first place -- to not have a second ball in storage override the rejection logic for the first ball.

Reply:
oop i thought i deleted that TODO but yes, i did follow that logic